beacon-chain/state/getters.go

```diff
@@ -224,6 +260,20 @@ func (b *BeaconState) Validators() []*ethpb.Validator {
 	return res
 }
 
+// ValidatorsNoClone returns validators participating in consensus on the beacon chain. This
+// method doesn't clone the respective validators.
+func (b *BeaconState) ValidatorsNoClone() []*Validator {
```

Comment:
hmm this is so confusing with the return type and *ethpb.Validator. Not clear to the reader what the difference is. Also, the naming can be improved, it seems unsafe from the outside but is not actually unsafe? Maybe we can rename this type to ValidatorWrapper? im not sure, maybe @prestonvanloon has some thoughts